### PR TITLE
Fix Dataflow SDK artifactId

### LINKED ISSUE #2181: "Could not retrieve available versions for Artifact com.google.cloud.dataflow" logged from Dataflow wizard
I see the following exception logged when going through the Dataflow wizard. I see it a couple more times at other points. However, nothing seems really broken about Dataflow lucikly.

The root cause:
```
java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/google/cloud/dataflow/com.google.cloud.dataflow/maven-metadata.xml
```

`https://repo1.maven.org/maven2/com/google/cloud/dataflow/` has several sub-folders, and `maven-metadata.xml` is under each sub-folder.

```
!ENTRY com.google.cloud.tools.eclipse.dataflow.core 2 0 2017-07-20 12:22:26.815
!MESSAGE Could not retrieve available versions for Artifact com.google.cloud.dataflow
!STACK 0
java.util.concurrent.ExecutionException: org.eclipse.core.runtime.CoreException: Could not retrieve Dataflow SDK Metadata
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:500)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:459)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:76)
	at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:142)
	at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2373)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2337)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2295)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2208)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4053)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4057)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4986)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever.getLatestIncrementalVersion(DataflowArtifactRetriever.java:153)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever.getLatestSdkVersion(DataflowArtifactRetriever.java:123)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowDependencyManager.getLatestDataflowDependencyInRange(DataflowDependencyManager.java:81)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowDependencyManager.getProjectMajorVersion(DataflowDependencyManager.java:160)
	at com.google.cloud.tools.eclipse.dataflow.ui.launcher.PipelineArgumentsTab.initializeFrom(PipelineArgumentsTab.java:300)
	at org.eclipse.debug.ui.AbstractLaunchConfigurationTabGroup.initializeFrom(AbstractLaunchConfigurationTabGroup.java:90)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupWrapper.initializeFrom(LaunchConfigurationTabGroupWrapper.java:201)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.displayInstanceTabs(LaunchConfigurationTabGroupViewer.java:766)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer$8.run(LaunchConfigurationTabGroupViewer.java:637)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.inputChanged(LaunchConfigurationTabGroupViewer.java:654)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.setInput0(LaunchConfigurationTabGroupViewer.java:615)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.setInput(LaunchConfigurationTabGroupViewer.java:590)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog.handleLaunchConfigurationSelectionChanged(LaunchConfigurationsDialog.java:960)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog$3.selectionChanged(LaunchConfigurationsDialog.java:573)
	at org.eclipse.jface.viewers.StructuredViewer$3.run(StructuredViewer.java:877)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.ui.internal.JFaceUtil$1.run(JFaceUtil.java:50)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
	at org.eclipse.jface.viewers.StructuredViewer.firePostSelectionChanged(StructuredViewer.java:874)
	at org.eclipse.jface.viewers.StructuredViewer.setSelection(StructuredViewer.java:1734)
	at org.eclipse.jface.viewers.TreeViewer.setSelection(TreeViewer.java:1093)
	at org.eclipse.jface.viewers.Viewer.setSelection(Viewer.java:388)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog.doInitialTreeSelection(LaunchConfigurationsDialog.java:613)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog.initializeContent(LaunchConfigurationsDialog.java:1101)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog.createContents(LaunchConfigurationsDialog.java:446)
	at org.eclipse.jface.window.Window.create(Window.java:430)
	at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1096)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog.create(LaunchConfigurationsDialog.java:374)
	at org.eclipse.jface.window.Window.open(Window.java:792)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog.open(LaunchConfigurationsDialog.java:1154)
	at org.eclipse.debug.ui.DebugUITools$2.run(DebugUITools.java:619)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.debug.ui.DebugUITools.openLaunchConfigurationDialogOnGroup(DebugUITools.java:627)
	at org.eclipse.debug.ui.DebugUITools.openLaunchConfigurationDialogOnGroup(DebugUITools.java:560)
	at org.eclipse.debug.ui.actions.OpenLaunchDialogAction.run(OpenLaunchDialogAction.java:82)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:473)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:595)
	at org.eclipse.jface.action.ActionContributionItem.access$2(ActionContributionItem.java:511)
	at org.eclipse.jface.action.ActionContributionItem$5.handleEvent(ActionContributionItem.java:420)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4481)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1329)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3819)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3430)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:694)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:606)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:139)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1515)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1488)
Caused by: org.eclipse.core.runtime.CoreException: Could not retrieve Dataflow SDK Metadata
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever.getMetadataDocument(DataflowArtifactRetriever.java:181)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever.access$0(DataflowArtifactRetriever.java:175)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever$2.load(DataflowArtifactRetriever.java:98)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever$2.load(DataflowArtifactRetriever.java:1)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3628)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2336)
	... 72 more
Caused by: java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/google/cloud/dataflow/com.google.cloud.dataflow/maven-metadata.xml
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1626)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(Unknown Source)
	at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:177)
	at com.google.cloud.tools.eclipse.dataflow.core.project.DataflowArtifactRetriever.getMetadataDocument(DataflowArtifactRetriever.java:179)
	... 77 more
```

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #2181

This test feels a bit… icky.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowDependencyManagerTest.java [hunk 1]
```diff
@@ -250,6 +255,22 @@ public void hasPinnedDependencyTrackedDependency() {
     assertFalse(manager.hasPinnedDataflowDependency(project));
   }
 
+  @Test
+  public void testGetLatestSdkVersion_validCoordinates() throws Exception {
+
+    when(artifactRetriever.getMetadataDocument(anyString()))
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowDependencyManagerTest.java]: Isn't this the default behavior?

**Comment by chanseokoh** [comment]
Test failure!

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowArtifactRetriever.java [hunk 2]
```diff
@@ -137,7 +138,8 @@ public ArtifactVersion getLatestArchetypeVersion(
    * Returns the latest Version of the specified Artifact in the version range, or null if there is
    * no such version.
    */
-  private ArtifactVersion getLatestIncrementalVersion(
+  @VisibleForTesting
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowArtifactRetriever.java]: medium term we need to replace this with ArtifactRetriever, which is tested

**Comment by briandealwis** [comment]
Tests pass if run outside of OSGi, and fail within.  I've seen this before and I don't understand the cause.

**Comment by elharo** [comment]
I'd be tempted to just drop the tests for now. We're going to replace this class soon anyway.

**Comment by briandealwis** [comment]
Backed out the test.

**Pull request merged**